categories.go

```diff
@@ -84,11 +84,25 @@ func newCategoriesFilterFromQuery(query url.Values) (*packages.Filter, error) {
 		}
 	}
 
+	// Deprecated: release tags to be removed.
 	if v := query.Get("experimental"); v != "" {
 		filter.Experimental, err = strconv.ParseBool(v)
 		if err != nil {
 			return nil, fmt.Errorf("invalid 'experimental' query param: '%s'", v)
 		}
+
+		// For compatibility with older versions of Kibana.
+		if filter.Experimental {
```

Comment:
Would you mind challenging this change a bit?

It looks like we're renaming `experimental=true` to `prerelease=true`. Meaning of both parameters is not so far from each other, so maybe we should deprecate it? It would not affect the API in terms of legacy support.

Reply:
Do you mean to remove `Experimental` from the filtering rules and use only `Prerelease`? I considered that, but decided to explicitly leave it because their implementations differ. I can reconsider it in any case, I think the resulting responses will be the same.

It will probably change the behaviour when no `experimental` flag is set, but this behaviour is also changed now.

(Btw, I am deprecating `experimental`, but it could still be used)

Reply:
Hm... what about staying with `experimental` instead of introducing a new flag? We could slightly extend its meaning to include more packages.

Reply:
Not a strong opinion, but in the future people will wonder why this flag is called `experimental` :slightly_smiling_face: 

Also, even when they have similar purpouses, they have slightly different behaviours. I prefer to include a new flag that future client implementations can use, and leave the old one as compatibility layer for old ones, that we can in some years directly remove when/if not used.

If the problem is about adding prerelease packages to the experimental response, and given that the only reason to do this is to maintain current behaviour for current implementations of kibana, that always use `experimental=true`. We can consider different alternatives:
* Version the API, so we add some prefix to endpoints, and we introduce the new behaviour only in the ones under `/v2`. This adds complexity, but we may end up needing something like this at some point in any case.
* Base the behaviour on the user agent. Future versions of Kibana are going to include their versions in the user agent (https://github.com/elastic/kibana/pull/121297), we could give different responses to different versions of Kibana. Though this is hacky and error prone, proxies may alter these headers.

Reply:
Got it 👍  Thanks for the discussion, let's try to preserve the KISS rule. I suppose that introducing another flag is simpler than preparing `/v2` endpoint.